Source/Boxed.DotnetNewTest/Framework/AssemblyResolver.cs

```diff
@@ -0,0 +1,67 @@
+namespace Boxed.Templates.FunctionalTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.Loader;
+    using Microsoft.Extensions.DependencyModel;
+    using Microsoft.Extensions.DependencyModel.Resolution;
+
+    internal sealed class AssemblyResolver : IDisposable
```

Comment:
I run functional tests by calling the dotnet CLI commands.

However, there is another way I have been experimenting with but have failed to get working so far. This works as follows:

1. Build the template project.
2. Load the built template assembly using the new `AssemblyLoadContext` API in .NET Core.
3. Uses the `TestHost` API in ASP.NET Core to run the API/Swagger/GraphQL/Orleans in memory.
4. Run tests code against it e.g. call the API running in memory.

In theory this will be a lot faster. The downside is that I can't yet get the assembly loading stuff to work reliably. Also .NET Core 3 has support for unloading assemblies at the end of the test which is also important.

To sum up, this code does not work but I want it to.

Reply:
I like this idea. Maybe its worth tabling this as a GitHub issue until core 3 is released in September?

Reply:
Yes, thats a good idea.